Dependencies/napi/source/js_native_api_JavaScriptCore.cc

```diff
@@ -464,6 +505,86 @@ namespace {
   };
 }
 
+struct napi_ref__ {
+  napi_ref__(napi_value value, uint32_t count)
+    : _value{value}
+    , _count{count} {
+  }
+  
+  napi_status init(napi_env env) {
+    // track the ref values to support weak refs
+    auto pair{env->active_ref_values.insert(_value)};
+    if (pair.second) {
+      ExternalInfo* info{};
+      CHECK_NAPI(ExternalInfo::Wrap(env, _value, &info));
+      info->AddFinalizer([iter{pair.first}](ExternalInfo* info) {
```

Comment:
So an interator for an `unordered_set` remains valid when the collection is mutated?

Reply:
Good catch. I was originally using a different container that was okay, but unordered_set is not okay.

https://en.cppreference.com/w/cpp/container/unordered_set
> Operations | Invalidated
> -- | --
> All read only operations, swap, std::swap | Never
> clear, rehash, reserve, operator= | Always
> insert, emplace, emplace_hint | Only if causes rehash
> erase | Only to the element erased
> 
>